### PR TITLE
ZEPPELIN-3218. Plugins for Interpreter Launcher

### PR DESCRIPTION
**Pull request by zjffdu**
### What is this PR for?
Move launcher into zeppelin plugins

### What type of PR is it?
[Refactoring]

### Todos
* [ ] - Task

### What is the Jira issue?
* https://issues.apache.org/jira/browse/ZEPPELIN-3218

### How should this be tested?
* CI pass

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? No

### CODE REVIEW

File: zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/InterpreterSetting.java [hunk 1]
```diff
@@ -665,6 +660,13 @@ public void clearNoteIdAndParaMap() {
     runtimeInfosToBeCleared = null;
   }
 
+  public String getLauncherPlugin() {
```
Comment (Leemoonsoo) [inline zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/InterpreterSetting.java]: Cool!  Do you have any idea how can we make InterpreterLauncher user configurable?

Comment (zjffdu) [reply zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/InterpreterSetting.java]: I was thinking to make it configurable in `interpreter-setting.json` and user could also customize it in UI. But I havn't nailed down the final design. Would do it in a separated PR after I complete the whole design.

Comment (felixcheung) [reply zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/InterpreterSetting.java]: group isn't guaranteed to be "spark" right?

Comment (zjffdu) [reply zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/InterpreterSetting.java]: It is always `spark` if it is spark interpreter even when there's multiple spark interpreters. We use field `name` to differentiate them

**Comment by zjffdu** [comment]
Will merge if no more comments

**Pull request closed without merging**